File: Level-2/롤케이크-자르기.py
from collections import Counter

def solution(topping):
    answer = 0
    be_ = set()
    af = Counter(topping)

    for t in topping:
        be_.add(t)
        af[t] -= 1

        if af[t] == 0:
            del af[t]
        if len(be_) == len(af):
            answer += 1

    return answer

File: Level-2/롤케이크-자르기.java
class Solution {
    public int solution(int[] topping) {
        int answer = 0;
        int[] left = new int[10001], right = new int[10001];
        int ls = 0, rs = 0;
        for(var i : topping) right[i]++;
        for(var i : right) rs += i > 0 ? 1 : 0;
        for(var i : topping) {
            right[i]--;
            if (right[i] == 0) rs--;
            if (left[i] == 0) ls++;
            left[i]++;
            if (rs == ls) answer++;
        }
        return answer;
    }
}
